server/src/main/java/org/elasticsearch/node/ResponseCollectorService.java

```diff
@@ -208,16 +206,12 @@ private static class NodeStatistics {
         final ExponentiallyWeightedMovingAverage responseTime;
         double serviceTime;
 
-        NodeStatistics(
-            String nodeId,
-            ExponentiallyWeightedMovingAverage queueSizeEWMA,
-            ExponentiallyWeightedMovingAverage responseTimeEWMA,
-            double serviceTimeEWMA
-        ) {
+        NodeStatistics(String nodeId, int queueSize, long responseTime, double serviceTimeEWMA) {
             this.nodeId = nodeId;
-            this.queueSize = queueSizeEWMA;
-            this.responseTime = responseTimeEWMA;
+            this.queueSize = new ExponentiallyWeightedMovingAverage(ALPHA, queueSize);
```

Comment:
Moving the object creation into the constructor here for readability + it's the cold path anyway.